src/fmu/dataio/dataio.py

```diff
@@ -87,75 +85,33 @@ def _validate_variable(key: str, value: type, legals: dict[str, str | type]) ->
-
-    elif isinstance(content, str):
-        logger.debug("content is a string")
-        if AllowedContent.requires_additional_input(content):
-            raise ValidationError(f"content {content} requires additional input")
-        usecontent = content
-        content_specific = None  # not relevant when content is a string
-        logger.debug("usecontent is %s", usecontent)
-
-    elif isinstance(content, dict):
-        logger.debug("content is a dictionary")
-        usecontent = (list(content.keys()))[0]
-        logger.debug("usecontent is %s", usecontent)
-        content_specific = content[usecontent]
-        logger.debug("content_specific is %s", content_specific)
-        if not isinstance(content_specific, dict):
-            raise ValueError(
-                "Content is incorrectly formatted. When giving content as a dict, "
-                "it must be formatted as:"
-                "{'mycontent': {extra_key: extra_value} where mycontent is a string "
-                "and in the list of valid contents, and extra keys in associated "
-                " dictionary must be valid keys for this content."
-            )
-    else:
-        raise ValidationError("The 'content' must be string or dict")
 
-    if usecontent != "unset" and usecontent not in AllowedContent.model_fields:
-        raise ValidationError(
-            f"Invalid content: <{usecontent}>! "
-            f"Valid content: {', '.join(AllowedContent.model_fields.keys())}"
-        )
+    content = deepcopy(content)
 
-    logger.debug("outgoing content is set to %s", usecontent)
-    if content_specific:
-        content_specific = _content_validate(usecontent, content_specific)
-    else:
-        logger.debug("content has no extra information")
+    if not content:
+        return AllowedContent(content="unset")
 
-    return usecontent, content_specific
+    if isinstance(content, str):
+        return AllowedContent(content=meta.enums.ContentEnum(content))
 
-
-def _content_validate(name: str, fields: dict[str, object] | None) -> dict | None:
-    try:
-        return AllowedContent.model_validate({name: fields}).model_dump(
-            exclude_none=True,
-            mode="json",
-        )[name]
-    except PydanticValidationError as e:
-        raise ValidationError(
-            f"""The field {name} has one or more errors that makes it
-impossible to create valid content. The data will still be exported but no
-metadata will be made. You are strongly encouraged to correct your
-configuration. Invalid configuration may be disallowed in future versions.
-
-Detailed information:
-{str(e)}
-"""
+    if len(content) > 1:
+        raise ValueError(
+            "Found more than one content item in the 'content' dictionary. Ensure "
+            "input is formatted as content={'mycontent': {extra_key: extra_value}}."
         )
 
+    usecontent, content_specific = list(content.items())[0]
```

Comment:
```suggestion
    usecontent, content_specific = next(iter(content.items()))
```

Just an idea, avoids the conversion to a list and avoids index syntax, but the difference is negligible

Reply:
agreed a bit easier on the eyes